### PR TITLE
Feature/user handling

### PR DESCRIPTION
**Pull request by danielronnkvist**
User and Identity model created.
- Functions for finding or creating both a user and a identity.
- Some helper methods for users handling, for example current_user.
- Model validations.
- Model testing, using Omniauth mockup.

### CODE REVIEW

**Comment by danielronnkvist** [comment]
Built from pull request #1

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -2,4 +2,11 @@ class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
+  helper_method :current_user
+
+  private
+
+  def current_user
+    @current_user ||= User.find(session[:user_id]) if session[:user_id]
```
Comment (danielronnkvist) [inline app/controllers/application_controller.rb:10]: Thank you Ryan!

http://railscasts.com/episodes/13-dangers-of-model-in-session

File: app/controllers/pages_controller.rb [hunk 2]
```diff
@@ -2,4 +2,8 @@ class PagesController < ApplicationController
   def show
     render "pages/#{params[:page]}"
   end
+
+  def omniauth_failure
```
Comment (klaseskilson) [inline app/controllers/pages_controller.rb:6]: Perhaps a message about what happened would be nice? Something like

``` ruby
flash[:alert] = "Could not link your account. :( Are you sure you gave us permission to do so?"
```

**Comment by klaseskilson** [comment]
Added flash messages to layout now. Looks good to me! Perhaps @eriol726 or @Fluffpannkaka could take a look so that we can merge this?

**Comment by ronjagrosz** [comment]
MERGE!

**Pull request merged**